scripts/utils/setup-roles.ts

```diff
@@ -0,0 +1,113 @@
+import { ethers } from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { ORIGINATOR_ROLE as DEFAULT_ORIGINATOR_ROLE, ADMIN_ROLE as DEFAULT_ADMIN_ROLE, FEE_CLAIMER_ROLE as DEFAULT_FEE_CLAIMER_ROLE, REPAYER_ROLE as DEFAULT_REPAYER_ROLE } from "./constants";
+
+
+export async function main (
+    VAULT_FACTORY_ADDRESS = "0x3A54241cB7801BDea625565AAcb0e873e79C0649",
+    LENDER_NOTE_ADDRESS = "0xdeaBbBe620EDF275F06E75E8fab18183389d606F",
+    BORROWER_NOTE_ADDRESS = "0xaEF48370a5f37CFb760CE44E6cbF986C4DeFF389",
+    ORIGINATION_CONTROLLER_ADDRESS = "0xFE046149bc8830d989A007E81A9D7F8A3b575F36",
+    LOAN_CORE_ADDRESS = "0xd624D1879429A606f54F48B08b56126c3Fe70049",
+    ADMIN_ADDRESS = "0x9b419fd36837558D8A3197a28a5e580AcE44f64F",
+    FEE_CONTROLLER_ADDRESS = "0xE4a1917Ebe8fd2CAFD79799C82aDAa7E81AC6D47",
+    REPAYMENT_CONTROLLER_ADDRESS = "0x23ce21bE3ebd1c86325100460D58d14a1D143E8d",
+    CALL_WHITELIST_ADDRESS = "0x8a12BB999100846B9E56aba4906762353C416952",
+    PUNK_ROUTER_ADDRESS= "0x76391cd8e269f2e8fDcf893E7F5E5781B2Fe2514",
+    ADMIN_ROLE = DEFAULT_ADMIN_ROLE,
+    FEE_CLAIMER_ROLE = DEFAULT_FEE_CLAIMER_ROLE,
+    ORIGINATOR_ROLE = DEFAULT_ORIGINATOR_ROLE,
+    REPAYER_ROLE = DEFAULT_REPAYER_ROLE
+): Promise<void> {
+    const signers: SignerWithAddress[] = await ethers.getSigners();
+    const [admin, adminMultiSig] = signers;
+    const deployer = admin;
+
+    if (!LOAN_CORE_ADDRESS) {
+        throw new Error("Must specify LOAN_CORE_ADDRESS in environment!");
+    }
+
+    if (!admin.address) {
+        throw new Error("Must specify ADMIN_ADDRESS in environment!");
+    }
+
+    if (FEE_CONTROLLER_ADDRESS) {
+        console.log("Fee controller address:", FEE_CONTROLLER_ADDRESS);
+    }
+
+    const loanCore = await ethers.getContractAt("LoanCore", LOAN_CORE_ADDRESS);
+    const vaultFactory = await ethers.getContractAt("VaultFactory", VAULT_FACTORY_ADDRESS);
+    const lenderNote = await ethers.getContractAt("PromissoryNote", LENDER_NOTE_ADDRESS);
+    const borrowerNote = await ethers.getContractAt("PromissoryNote", BORROWER_NOTE_ADDRESS);
+
+    // grant correct permissions for promissory note
+    // giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
+        await note.connect(await admin).initialize(loanCore.address);
+    }
+
+    // grant LoanCore admin fee claimer permissions
+    const updateLoanCoreFeeClaimer = await loanCore.connect(admin).grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreFeeClaimer.wait();
+
```

Comment:
After each of these tx's runs can we add a console.log to show that it was successful? When I ran this it took like 30 secs to 1 min and I was sort of unsure where in the process it was.